File: factorial/factorial.c
#include <stdio.h>

double factorial(int n) {
    double value = 0.0;

    if (n == 0 || n == 1) {
        value = 1;
    } else {
        value = n * factorial(n - 1);
    }

    return (value);
}

int main(int argc, char *argv[]) {
    int input = 0;

    printf("** Compute Factorial **\n");
    printf("Enter An Integer: ");
    scanf("%d", &input);

    if (input >= 0 && input <= 170) {
        double f = factorial(input);
        printf("%d! = %.0f\n", input, f);
    } else {
        printf("Sorry! Can't compute !%d.\n", input);
    }

    return 0;
}

File: factorial/factorial.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

class Factorial {

    private static double factorial(int n) {
        double value = 0.0;

        if (n == 0 || n == 1) {
            value = 1;
        } else {
            value = n * factorial(n - 1);
        }

        return (value);
    }

    public static void main(String args[]) throws Exception {
        String input = null;

        System.out.println("** Compute Factorial **");
        System.out.print("Enter An Integer: ");

        BufferedReader inputReader = new BufferedReader(new InputStreamReader(System.in));
        input = inputReader.readLine();

        int n = Integer.parseInt(input.trim());

        if (n >= 0 && n <= 170) {
            double f = factorial(n);
            System.out.printf("%d! = %.0f\n", n, f);
        } else {
            System.out.printf("Sorry! Can't compute !%d.\n", n);
        }
    }
}
